packages/chat-component/src/main.ts

```diff
@@ -187,6 +195,49 @@ export class ChatComponent extends LitElement {
     this.currentQuestion = this.questionInput.value;
   }
 
+  async handleCitationClick(sourceUrl: string, event: Event): Promise<void> {
+    if (sourceUrl?.endsWith('.md')) {
+      event?.preventDefault();
+
+      if (!this.isShowingThoughtProcess) {
+        this.selectedAsideTab = 'tab-citations';
+        this.showThoughtProcess();
+      }
+
+      const response = await fetch(sourceUrl);
+      if (response.ok) {
+        // highlight the clicked citation to make it clear which is being previewed
+        const citationsList = this.shadowRoot?.querySelectorAll(
+          '.aside .items__list.citations .items__listItem--citation',
+        );
+
+        if (citationsList) {
+          const citationsArray = [...citationsList];
+          const clickedIndex = citationsArray.findIndex((citation) => {
+            const link = citation.querySelector('a');
+            return link?.href === sourceUrl;
+          });
+
+          for (const citation of citationsList) {
+            const index = citationsArray.indexOf(citation);
+            if (index === clickedIndex) {
+              citation.classList.add('active');
```

Comment:
instead of directly modifying the DOM why not use Lit reactive state instead?

Reply:
This is beyond the scope of this PR, since this code was already in. Please review the preview feature as we will optimize for lit in the component refactoring.

Reply:
My bad, I didn't check that it was there before.